### PR TITLE
CHROMEOS rootfs builds implementation

### PR DESCRIPTION
**Pull request by nuclearcat**
To implement ChromiumOS builds in KernelCI jenkins several changes required.
Details of changes in relevant commit messages

Signed-off-by: Denys Fedoryshchenko <denys.f@collabora.com>

### CODE REVIEW

**Comment by nuclearcat** [comment]
Some tests and discussion is in old PR: https://github.com/kernelci/kernelci-jenkins/pull/77

**Comment by gctucker** [comment]
Please change the base branch to `chromeos`

**Comment by nuclearcat** [comment]
> Please change the base branch to `chromeos`
Done

**Comment by nuclearcat** [comment]
Job: https://bot.staging.kernelci.org/job/denysf/job/rootfs-builder-denysf/39/console

https://storage.chromeos.kernelci.org/images/rootfs/chromeos/crostest005/chromiumos-amd64-generic/amd64/
Path probably too long, by next patch i can shrink it, or maybe i should edit old ones?

I will do one more test with existing directory, adding || true in mkdir, so it wont fail middle of the upload if directory already exist.

**Comment by nuclearcat** [comment]
Tested, works fine
https://bot.staging.kernelci.org/job/denysf/job/rootfs-builder-denysf/40/console

File: jobs/rootfs-builder.jpl [hunk 1]
```diff
@@ -106,18 +115,42 @@ node("debos && docker") {
...
             }
         }
 
+        stage("scp") {
+            sshagent(credentials : ['jenkins-chromeos-storage-private-key']) {
+                sh(script: """\
+ssh \
+-o StrictHostKeyChecking=no \
+-p 22022 \
+chromeos@storage.chromeos.kernelci.org 'mkdir ~/images/rootfs/chromeos/${pipeline_version} || true'
```
Comment (gctucker) [inline jobs/rootfs-builder.jpl:132]:
```suggestion
chromeos@storage.chromeos.kernelci.org 'mkdir -p ~/images/rootfs/chromeos/${pipeline_version} || true'
```
to solve issues like this:
```
mkdir: cannot create directory '/home/chromeos/images/rootfs/chromeos/crostest005': File exists
```

File: jobs/rootfs-builder.jpl [hunk 2]
```diff
@@ -123,10 +123,34 @@ node("debos && docker") {
...
+            }
+        }
+/* Upload stage not working, as kci_rootfs http(s) upload fails for files >1GB */
+/*
         stage("Upload") {
             timeout(time: 30, unit: 'MINUTES') {
                 upload(config, pipeline_version, kci_core, rootfs_type, arch);
             }
         }
+*/
```
Comment (gctucker) [inline jobs/rootfs-builder.jpl:154]: I think we can just delete this now rather than keep unused code in the file.

File: jobs/rootfs-builder.jpl [hunk 3]
```diff
@@ -123,10 +123,34 @@ node("debos && docker") {
             }
         }
 
+        stage("scp") {
```
Comment (gctucker) [inline jobs/rootfs-builder.jpl:126]: Maybe `stage("Upload")` would be a bit better, to keep the same name as before. That way, all the roofs-builder jobs will appear with the same pipeline steps in Jenkins.

File: jobs/rootfs-builder.jpl (hunk 2 above)
Comment (nuclearcat) [reply jobs/rootfs-builder.jpl]: Done

File: jobs/rootfs-builder.jpl (hunk 1 above)
Comment (nuclearcat) [reply jobs/rootfs-builder.jpl]: Having -p is great idea, i missed that, thanks, added.

**Review by gctucker — APPROVED** [review]
OK thanks, the last changes were trivial so I don't think it's worth doing another build just to confirm it's still working.

**Pull request merged**